### PR TITLE
Clean up path handling for push

### PR DESCRIPTION
**Pull request by techtonik**
This removes system dependent path manipulations (path/filepath) as target path uses forward slashes always. A better fix for #2565.

### CODE REVIEW

**Comment by techtonik** [comment]
AppVeyor failed to download some packages.

**Comment by stgraber** [comment]
Yeah, appveyor does that a lot...

**Comment by techtonik** [comment]
Maybe enabling proxy may help - https://www.appveyor.com/docs/how-to/http-proxy/

**Comment by stgraber** [comment]
If git was using http, it might, but since it's over its own protocol or over https, that's unlikely to help much.

**Comment by stgraber** [comment]
I'll leave that one for @tych0 to review since he's the one who knows how that stuff is supposed to work.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by techtonik** [comment]
Yea. right. Anyway, tests for this specific issue are only in end to end tests, so AppVeyor doesn't test it. Maybe still move path normalization so that it could be tested?

File: lxc/file.go [hunk 1]
```diff
@@ -134,9 +138,7 @@ func (c *fileCmd) push(config *lxd.Config, send_file_perms bool, args []string)
 		gid = c.gid
 	}
 
-	_, targetfilename := filepath.Split(targetPath)
-
-	if (targetfilename != "") && (len(sourcefilenames) > 1) {
+	if (len(sourcefilenames) > 1) && !targetIsDir {
```
Comment (techtonik) [inline lxc/file.go]: This probably is the most important part.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by tych0** [comment]
I'm confused. Isn't this basically just a revert of the normalize commit? What was the real bug that was fixed?

**Comment by tych0** [comment]
Oh, no, I see now. LGTM

**Comment by stgraber** [comment]
This branch breaks "lxc file push -p source/foo filemanip/.". Causes:
```
error: truncate: Invalid argument
errno: 22
error: copy: Invalid argument
errno: 22
```

**Comment by techtonik** [comment]
I wish there could be a better error message than `truncate: Invalid argument`. Anyway, the problem was in `_, targetfilename := filepath.Split(targetPath)`. For `/.` case my check for isDir failed, because I do check before normalization. `Split` makes check after normalization. The solution is to have a double check before and after normalization.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by techtonik** [comment]
migration test failed. I wonder what does it mean.. Is there any plugin for Jenkins for more structured test output?

**Comment by stgraber** [comment]
It's a single seemingly unrelated test failure, that won't hold this branch.

@tych0 can you take a look at the changes that went in here before we hit merge?

**Comment by techtonik** [comment]
@tych0 ping?

**Comment by tych0** [comment]
So now this is basically a full revert of the commit :). Anyway, fine with me.

**Pull request merged**